src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/feerate.h>
+#include <primitives/transaction.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
+#include <wallet/coinselection.h>
+
+#include <vector>
+
+namespace wallet {
+
+static void AddCoin(const CAmount& value, const int n_input, const int n_input_bytes, const int locktime, std::vector<COutput>& coins)
+{
+    CMutableTransaction tx;
+    tx.vout.resize(n_input + 1);
+    tx.vout[n_input].nValue = value;
+    tx.nLockTime = locktime; // all transactions get different hashes
+    coins.emplace_back(COutPoint(tx.GetHash(), n_input), tx.vout.at(n_input), /*depth=*/0, n_input_bytes, /*spendable=*/true, /*solvable=*/true, /*safe=*/true, /*time=*/0, /*from_me=*/true);
+}
+
+// Randomly distribute coins to instances of OutputGroup
+static void GroupCoins(FuzzedDataProvider& fuzzed_data_provider, const std::vector<COutput>& coins, const CoinSelectionParams& coin_params, const bool positive_only, std::vector<OutputGroup>& output_groups)
+{
+    auto output_group = OutputGroup(coin_params);
+    bool valid_outputgroup{false};
+    for (auto& coin : coins) {
+        output_group.Insert(coin, /*ancestors=*/0, /*descendants=*/0, positive_only);
+        // If positive_only was specified, nothing may have been inserted, leading to an empty outpout group
+        // that would be invalid for the BnB algorithm
+        valid_outputgroup = !positive_only || output_group.GetSelectionAmount() > 0;
+        if (valid_outputgroup && fuzzed_data_provider.ConsumeBool()) {
+            output_groups.push_back(output_group);
+            output_group = OutputGroup(coin_params);
+            valid_outputgroup = false;
+        }
+    }
+    if (valid_outputgroup) output_groups.push_back(output_group);
+}
+
+FUZZ_TARGET(coinselection)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    std::vector<COutput> utxo_pool;
+
+    const CFeeRate long_term_fee_rate{CFeeRate(ConsumeMoney(fuzzed_data_provider, /*max=*/COIN))};
```

Comment:
nit: One constructor should be sufficient.

```suggestion
    const CFeeRate long_term_fee_rate{ConsumeMoney(fuzzed_data_provider, /*max=*/COIN)};
```

Reply:
Yes, copy/paste mistake, corrected.